crates/bevy_reflect/src/type_name.rs

```diff
@@ -1,15 +1,58 @@
 use crate::utility::GenericTypeNameCell;
 
-/// Provide the name of the type as string.
-pub trait TypeName {
+/// Provide the name of the type as a string slice.
```

Comment:
Nit: You can probably just refer to it as a `string` rather than a `string slice`.